File: BOJ/SeokhyeonHeo/스택큐/BOJ_10828.py
import sys

N = int(sys.stdin.readline())

Stack = []

for _ in range(N):
    c = sys.stdin.readline().split()
    if c[0] == 'push':
        Stack.append(c[1])
    elif c[0] =='pop':
        if Stack:
            print(Stack.pop())
        else:
            print(-1)
    elif c[0] == 'size':
        print(len(Stack))
    elif c[0] == 'empty':
        if Stack:
            print(0)
        else:
            print(1)
    elif c[0] == 'top':
        if Stack:
            print(Stack[-1])
        else:
            print(-1)
        

File: BOJ/JeongHyeYeon/스택큐/BOJ_10828.js
const array = require("fs").readFileSync("/dev/stdin").toString().split("\n");
// const array = ['7', 'pop', 'top', 'push 123', 'top', 'pop', 'top', 'pop'];

const stack = [];
const result = [];

const length = array.shift();

for (i = 0; i < length; i++) {
  switch (array[i]) {
    case "pop":
      result.push(stack.length == 0 ? -1 : stack.pop());
      break;

    case "size":
      result.push(stack.length);
      break;

    case "empty":
      result.push(stack.length == 0 ? 1 : 0);
      break;

    case "top":
      result.push(stack.length == 0 ? -1 : stack[stack.length - 1]);
      break;

    default:
      stack.push(array[i].split(" ")[1]);
      break;
  }
}

console.log(result.join("\n"));
